documentation/src/docs/security/generating-tls-artifacts.md

```diff
@@ -0,0 +1,324 @@
+reverse proxy, Kubernetes Ingress or a load balancer. If TLS is terminated at such an infrastructural component, instead 
+of Pravega services, you will need to generate TLS artifacts for those infrastructural components instead, using similar
+steps described in this document; however, the exact steps may be slightly different and additional steps may be required to 
+ensure that other parts of the infrastructure can communicate with it. Refer to the respective platform vendor's 
+documentation, 
+such as [this](https://cloud.google.com/kubernetes-engine/docs/concepts/ingress) one from Google Kubernetes Engine, 
+for instructions.
+
+**Before you Begin:**
+
+The steps described in this document use OpenSSL and Java Keytool to generate TLS artifacts like certificates, keys, 
+Java keystores and Java truststores. Install OpenSSL and Java Development Kit (JDK) on the hosts where the artifacts 
+will be generated.
+
+**Note:**
+* The examples shown in this document use command line arguments to pass all inputs to the command. To pass
+  sensitive command arguments via prompts instead, just exclude the corresponding option (as shown in the second command
+  below). 
+
+  ```bash
+  # Passwords are passed as command line arguments.
+  $ keytool -keystore server01.keystore.jks -alias server01 -validity <validity> -genkey \
+              -storepass <keystore-password> -keypass <key-password> \
+              -dname <distinguished-name> -ext SAN=DNS:<hostname>,
+
+  # Passwords (and some other arguments) are to be entered interactively on the prompt.
+  $ keytool -keystore server01.keystore.jks -alias server01 -genkey
+  ```
+* A weak password `changeit` is used everywhere in this document for easier reading. Be sure to replace it with 
+  a strong and separate password for each file.
+
+## Setting up a Certificate Authority (CA)
+
+If you are planning to use a public CA or an existing private/internal CA, you may skip 
+this part altogether and go directly to the [next section](#generate-tls-certificates-and-other-tls-artifacts-for-the-server).
+
+Here, we'll create a CA in the form of a public/private key pair and a self-signed certificate.
+Later, we'll use this CA certificate/key bundle to sign the servers' certificates.
+
+1. Generate a CA certificate and public/private key pair. 
+
+   ```bash
+   $ openssl req -new -x509 -keyout <ca-private-key-file-path> -out <ca-cert-file-path> -days <validity-in-days> \
+            -subj "<distinguished_name>" \
+            -passout pass:<ca-private-key-password>
+   ```
+
+   Example:
+   ```bash
+   $ openssl req -new -x509 -keyout ca-key.key -out ca-cert.crt -days 365 \
+            -subj "/C=US/ST=Washington/L=Seattle/O=Pravega/OU=CA/CN=Pravega-Stack-CA" \
+            -passout pass:changeit
+   ```
+  
+   The command above will generate the following two PEM-encoded files: 
+   * A file containing the encrypted private key.
+   * A file containing the CA certificate.
```

Comment:
Consider listing filenames, `ca-key.key` and `ca-cert.crt`. It should be obvious, but just to be explicit.

Reply:
Done.